xbmc/music/windows/GUIWindowVisualisation.cpp

```diff
@@ -240,3 +241,29 @@ void CGUIWindowVisualisation::FrameMove()
   }
   CGUIWindow::FrameMove();
 }
+
+void CGUIWindowVisualisation::ToggleOSD()
+{
+  CGUIDialogMusicOSD *pOSD = static_cast<CGUIDialogMusicOSD *>(g_windowManager.GetWindow(WINDOW_DIALOG_MUSIC_OSD));
+  if (pOSD)
+  {
+    if (pOSD->IsDialogRunning())
+    {
+      pOSD->Close();
+    }
+    else if (g_infoManager.GetShowInfo() &&
+             CSettings::Get().GetBool("pvrplayback.confirmchannelswitch") &&
+             g_application.CurrentFileItem().HasPVRChannelInfoTag())
+    {
+      // If only channel info but no player controls are currently visible
+      // and setting "confirm channel switch with OK" is active, do a channel
+      // switch instead of turning on the the player controls.
+      g_application.m_pPlayer->SwitchChannel(g_application.CurrentFileItem().GetPVRChannelInfoTag());
```

Comment:
As I said before, best we introduce a new dialog which becomes active if you switch a channel (tv/radio) and display the related channel info. Within this dialog you can handle the action `ACTION_SELECT_ITEM` and do the necessary stuff. There is no way to handle what you are currently want to achieve within the FullscreenVideo/Visualisation window because the OK/Enter key is mapped differently depends on the use-case. Keep in mind that you need to handle channel up/down in this new dialog too. There is no fallback to fullscreen/visualisation window's key mapping. Another advantage you can cancel/close this dialog (hide the channel info) with back, which is also not possible in the current implementation. You need to wait until the info control disappears.